test/test_torch.py

```diff
@@ -461,6 +461,22 @@ def test_all_any_empty(self):
         self.assertTrue(x.all())
         self.assertFalse(x.any())
 
+    @unittest.skipIf(not TEST_NUMPY, "Numpy not found")
+    def test_all_any_with_dim(self):
+        def test(x):
+            r1 = np.all(x, axis=1, keepdims=False).astype('uint8')
```

Comment:
Can you do `x.float().sum() > 0` for OR, and `== x.dim(dim)` for AND? That would remove the numpy dependency

Reply:
I have fixed the test using `torch.sum` and `torch.prod`, removing the numpy dependency.